ext/standard/exec.c

```diff
@@ -542,8 +546,11 @@ PHP_FUNCTION(shell_exec)
 
 	stream = php_stream_fopen_from_pipe(in, "rb");
 	ret = php_stream_copy_to_mem(stream, PHP_STREAM_COPY_ALL, 0);
-	php_stream_close(stream);
+	ret_pclose = php_stream_close(stream);
 
+	if (ret_code) {
```

Comment:
Won't this evaluate to `false` if `ret_code === 0`?

Reply:
If so, is `exec` not also affected by this? https://github.com/php/php-src/blob/41d2785f4d8e069300115f9f16d18eef607488a8/ext/standard/exec.c#L246-L248

Reply:
TL;DR: this code is fine.
[First, that's C not PHP, so there's no "`===`" operator; second,] `ret_code` is a `zval *` i.e. a *pointer* to `zval`, so `if (ret_code)` means `if (ret_code != NULL)` i.e. (here) the PHP function was called with a 2nd argument (a variable into which to copy the value of `ret_pclose`).